File: frontend/src/service/sales.service.js
import axios from "axios";

const API_URL = "http://localhost:8080"; 

class SalesService {

    saveSales(sales) {
        return axios.post(API_URL + "/saveSales", sales);
    }

    getAllSales() {
        return axios.get(API_URL + "/getAllSales");
    }

    getSalesById(id) {
        return axios.get(API_URL + "/getSalesById" + id);
    }

    deleteSales(id) {
        return axios.get(API_URL + "/deleteSales/" + id);
    }

    editSales(sales) {
        return axios.post(API_URL + "/editSales/" + sales.id, sales);
    }

}

export default new SalesService;

File: src/main/java/com/example/inventorymanagement/service/SalesService.java
package com.example.inventorymanagement.service;

import com.example.inventorymanagement.model.Sales;
import java.util.List;

public interface SalesService {
    public Sales saveSales(Sales sales);
    public List<Sales> getAllSales();
    public Sales getSalesById(Integer id);
    public String deleteSales(Integer id);
    public Sales editSales(Sales sales, Integer id);
}
